core/blockchain.go

```diff
@@ -666,6 +666,8 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 			PutTransactions(self.chainDb, block, block.Transactions())
 			// store the receipts
 			PutReceipts(self.chainDb, receipts)
+			// Write map map bloom filters
+			WriteMipmapBloom(self.chainDb, block.NumberU64(), receipts)
```

Comment:
If you put it here, you can lose data.

Eg: You have an existing empty chain with no receipts: A-B-C-D, and a fork comes in rooted at B: A-B-1-X-Y. The `1` contains a receipt, none of the others do. When you insert `1`, this bloom write doesn't get called, since 1 is only a side chain at that point. Then you insert X and Y, your chain gets reorged, but the bloom is never updated any more. So now you've lost knowledge about the receipt in `1`.